packages/otel/src/index.ts

```diff
@@ -0,0 +1,59 @@
+import { SpanKind, SpanStatusCode, type TracerProvider, trace } from '@opentelemetry/api'
+import {
+  ATTR_HTTP_REQUEST_HEADER,
+  ATTR_HTTP_REQUEST_METHOD,
+  ATTR_HTTP_RESPONSE_HEADER,
+  ATTR_HTTP_RESPONSE_STATUS_CODE,
+  ATTR_URL_FULL,
+  ATTR_HTTP_ROUTE,
+} from '@opentelemetry/semantic-conventions'
+import { createMiddleware } from 'hono/factory'
+import type { Env, Input } from 'hono'
+
+const PACKAGE_NAME = '@hono/otel'
+const PACKAGE_VERSION = '0.1.0'
+
+export interface OtelOptions {
+  tracerProvider?: TracerProvider
+}
+
+export const otel = <E extends Env = any, P extends string = any, I extends Input = {}>(
+  options: OtelOptions = {}
+) => {
+  const tracerProvider = options.tracerProvider ?? trace.getTracerProvider()
+  return createMiddleware<E, P, I>(async (c, next) => {
+    const tracer = tracerProvider.getTracer(PACKAGE_NAME, PACKAGE_VERSION)
```

Comment:
You should do this outside of `createMiddleware()` (right after `tracerProvider`).

Reply:
Fixed!